core/src/main/java/io/parsingdata/metal/expression/value/reference/First.java

```diff
@@ -34,7 +35,11 @@ public First(final ValueExpression op) {
     @Override
     public OptionalValueList eval(final Environment env, final Encoding enc) {
         final OptionalValueList list = _op.eval(env, enc);
-        return list.isEmpty() ? list : OptionalValueList.create(list.reverse().head);
+        return list.isEmpty() ? list : OptionalValueList.create(getFirst(list));
```

Comment:
It's the implementation of `First`. Tests for it were added when it was introduced. For example, see tests 13-15, 23-27 and 29-30 in `ReferenceValueExpressionSemanticsTest`.